### PR TITLE
[specifications] Flesh out the description of Execution in the Move Adapter

### PR DESCRIPTION
**Pull request by bob-wilson**
This also changes a few comments to clarify things that I noticed while
working on the specification. For the VM section of the specification,
I updated a few code signatures to match the v1 code, but the rest of those
changes are just formatting and typo fixes.

## Motivation

More complete specification of the Move adapter

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

File: specifications/move_adapter/README.md [hunk 1]
```diff
@@ -308,86 +327,96 @@ pub enum WriteOp {
...
+
+The Move VM operations used here consume gas according to the gas schedule that is stored in an on-chain configuration. The adapter loads that gas schedule before invoking the VM.
+
+If the script or module payload is processed successfully, the Move VM is next used to [execute](#Function-Execution) the `epilogue` function from the `LibraAccount` module. The epilogue increments the sender's `sequence_number` and deducts the transaction fee based on the gas price and the amount of gas consumed. This function execution is done using the same VM `Session` that was used when processing the payload, so that all the side effects are combined. The epilogue function is run with gas metering disabled.
+
+On the other hand, if an error occurs when processing the payload, the adapter will still run the epilogue function, but it will do so in a new VM `Session` so that any side effects from the payload will not be saved. This has the effect of charging the transaction fee for gas consumption to the sender's account.
+
+If an error occurs while running the epilogue function, then the transaction is discarded.
```
Comment (dariorussi) [inline specifications/move_adapter/README.md:409]: is this accurate? I thought the epilogue was error free?

File: specifications/move_adapter/README.md [hunk 2]
```diff
@@ -866,135 +912,139 @@ has the following entries:
...
+    * **_struct handle_**: ULEB128 - index into the STRUCT\_HANDLES table of the generic struct for
     this instantiation (e.g. `struct X<T>`)
     * **_instantiation_**: ULEB128 - index into the SIGNATURES table for the instantiation of the struct
     The instantiation can be either partial or complete (e.g. `X<U64>` or `X<Z>` when inside another generic
     function or generic struct with type parameter `Z`)
-* **_FUNCTION_DEFINITIONS_** - the set of functions defined in this binary. A function definition contains the
+* **_FUNCTION\_DEFINITIONS_** - the set of functions defined in this binary. A function definition contains the
 following fields:
-    * **_function_handle_**: ULEB128 - index in the FUNCTION_HANDLES table for the handle of this definition
+    * **_function\_handle_**: ULEB128 - index in the FUNCTION\_HANDLES table for the handle of this definition
```
Comment (dariorussi) [inline specifications/move_adapter/README.md:937]: wow to all of those....

**Review by dariorussi — APPROVED** [review]
LGTM. Like the spec changes! thanks

File: specifications/move_adapter/README.md [hunk 3]
```diff
@@ -308,86 +327,96 @@ pub enum WriteOp {
...
+
+* `UserTransaction`: The first step here is to repeat the transaction validation process in case things have changed since the initial validation. See the [Validation section](#Validation) for details. Invalid transactions are discarded. The signature verification step of validation is computationally expensive, so for performance reasons, it is a good idea for the adapter implementation to do that checking in parallel for all the input transactions, instead of waiting to do it as the transactions are executed sequentially. After successful validation, the `TransactionPayload` is processed as described in the following sections for different kinds of payloads.
```
[the remark this answers is not in the document]
Comment (bob-wilson) [reply specifications/move_adapter/README.md:385]: I'm not sure how to interpret your comment. I wanted to call this out but also make clear that it's an implementation detail, not necessarily part of the specified behavior.

File: specifications/move_adapter/README.md (hunk 1 above)
Comment (bob-wilson) [reply specifications/move_adapter/README.md:409]: Errors are not expected but the epilogue can still in theory fail with `UNEXPECTED_ERROR_FROM_KNOWN_MOVE_FUNCTION`

File: specifications/move_adapter/README.md (hunk 2 above)
Comment (bob-wilson) [reply specifications/move_adapter/README.md:937]: If you put them in backquotes (as code), then you don't generally need to escape the underscores. I assumed that you chose to write them this way on purpose, though

File: specifications/move_adapter/README.md (hunk 1 above)
Comment (bob-wilson) [reply specifications/move_adapter/README.md:409]: Somehow I also lost the description of the case where the script succeeds but the epilogue fails (e.g., sent payment causes insufficient balance to pay gas). I will fix that, too

**Comment by bob-wilson** [comment]
@dariorussi I'll go ahead and land this to unblock any subsequent edits that you are working on, but please take a look at the revised paragraph about the epilogue to make sure I got it right this time

**Pull request closed without merging**